coreneuron/gpu/nrn_acc_manager.cpp

```diff
@@ -626,44 +626,50 @@ void realloc_net_receive_buffer(NrnThread* nt, Memb_list* ml) {
         cnrn_target_delete(nrb->_nrb_index, nrb->_size);
     }
 #endif
-
-    // Reallocate host
+    // Reallocate host buffers using ecalloc_align (as in phase2.cpp) and
+    // free_memory (as in nrn_setup.cpp)
+    auto const realloc = [old_size = nrb->_size, nrb](auto*& ptr, std::size_t extra_size = 0) {
+        using T = std::remove_pointer_t<std::remove_reference_t<decltype(ptr)>>;
+        static_assert(std::is_trivial<T>::value,
+                      "Only trivially constructible and copiable types are supported.");
+        static_assert(std::is_same<decltype(ptr), T*&>::value,
+                      "ptr should be reference-to-pointer");
+        auto* const new_data = static_cast<T*>(ecalloc_align((nrb->_size + extra_size), sizeof(T)));
+        std::memcpy(new_data, ptr, (old_size + extra_size) * sizeof(T));
+        free_memory(ptr);
+        ptr = new_data;
+    };
```

Comment:
could remind me if this is done for consistency and to use unified memory in case unified-memory-build?

Reply:
It's for consistency. The unified memory builds are a more extreme example, where this would have called `realloc` on a unified memory pointer, but even in standard builds I think this was pairing `realloc` with `posix_memalign`, which seems dubious (https://linux.die.net/man/3/realloc).